src/utils/haulPortHandler.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2017-present, Callstack.
+ * All rights reserved.
+ *
+ * @flow
+ */
+
+const net = require('net');
+const exec = require('child_process').exec;
+
+/*
+ * Check if the port is already in use
+ */
+function isPortTaken(port: number) {
+  return new Promise(resolve => {
+    const portTester = net
+      .createServer()
+      .once('error', () => {
```

Comment:
Should we check if `err.code === 'EADDRINUSE'`

Reply:
First error we'll get when starting a server is `EADDRINUSE`, so I think we can safely skip this